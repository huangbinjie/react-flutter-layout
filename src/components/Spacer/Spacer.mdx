---
name: Spacer
menu: components
---

import { Playground, Props } from 'docz'
import Container from '../Container'
import Spacer from '../Spacer'
import Row from '../Row'

## Spacer

占满剩余空间，等价于 `flex: 1`

### Basic

<Playground>
  <Row>
    <Container width="100px" height="100px" color="#ececec" />
    <Spacer />
    <Container width="100px" height="100px" color="coral" />
  </Row>
</Playground>

## Props

无
