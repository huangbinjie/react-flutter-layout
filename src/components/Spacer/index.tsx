import React from 'react'

export default function Spacer() {
  return <div debug-label="Spacer" style={{ flex: 1 }}></div>
}
