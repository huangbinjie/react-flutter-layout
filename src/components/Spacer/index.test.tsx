import React from 'react'
import renderer from 'react-test-renderer'

import Spacer from '.'

describe('Spacer', () => {
  test('basic', () => {
    const tree = renderer.create(
      <Spacer />
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
