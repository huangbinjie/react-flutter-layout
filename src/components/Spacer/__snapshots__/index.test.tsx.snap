// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spacer basic 1`] = `
<div
  debug-label="Spacer"
  style={
    Object {
      "flex": 1,
    }
  }
/>
`;
